homeassistant/components/fritzbox/sensor.py

```diff
@@ -26,7 +24,7 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the FRITZ!SmartHome sensor from ConfigEntry."""
-    entities = []
+    entities: list[FritzBoxBatterySensor | FritzBoxTempSensor] = []
```

Comment:
It's better to use base class.
```suggestion
    entities: list[FritzBoxEntity] = []
```